### PR TITLE
Extend hp support in FEInterfaceValues

### PR DESCRIPTION
**Pull request by jppelteret**
Part of #14594

### CODE REVIEW

File: include/deal.II/fe/fe_interface_values.h [hunk 1]
```diff
@@ -1479,6 +1479,21 @@ class FEInterfaceValues
...
+   * @param[in] mapping_index If left at its default, use the mapping within the
+   * hp::MappingCollection passed to the constructor as given by the dominating
+   * finite element across the interface. (Only used if the FEInterface object
+   * is initialised with an hp::FECollection, an hp::QCollection and possibly an
+   * hp::MappingCollection.)
+   * @param[in] fe_index If left at its default, use the finite element within
+   * the hp::FECollection passed to the constructor as given by the dominating
+   * finite element across the interface. (Only used if the FEInterface object
+   * is initialised with an hp::FECollection, an hp::QCollection and possibly an
+   * hp::MappingCollection.)
```
Comment (luca-heltai) [inline include/deal.II/fe/fe_interface_values.h:1496]:
```suggestion
   * dominating finite element across the interface (only used if the
   * FEInterface object is initialised with an hp::FECollection, an
   * hp::QCollection, and possibly an hp::MappingCollection).
   * @param[in] mapping_index If left at its default, use the mapping within the
   * hp::MappingCollection passed to the constructor as given by the dominating
   * finite element across the interface (only used if the FEInterface object
   * is initialised with an hp::FECollection, an hp::QCollection, and possibly an
   * hp::MappingCollection).
   * @param[in] fe_index If left at its default, use the finite element within
   * the hp::FECollection passed to the constructor as given by the dominating
   * finite element across the interface (only used if the FEInterface object
   * is initialised with an hp::FECollection, an hp::QCollection, and possibly an
   * hp::MappingCollection).
```

Comment (luca-heltai) [inline include/deal.II/fe/fe_interface_values.h:1496]:
```suggestion
   * dominating finite element across the interface (only used if the
   * FEInterface object is initialised with an hp::FECollection, an
   * hp::QCollection, and possibly an hp::MappingCollection).
   * @param[in] mapping_index If left at its default, use the mapping within the
   * hp::MappingCollection passed to the constructor as given by the dominating
   * finite element across the interface (only used if the FEInterface object
   * is initialised with an hp::FECollection, an hp::QCollection, and possibly an
   * hp::MappingCollection).
   * @param[in] fe_index If left at its default, use the finite element within
   * the hp::FECollection passed to the constructor as given by the dominating
   * finite element across the interface (only used if the FEInterface object
   * is initialised with an hp::FECollection, an hp::QCollection, and possibly an
   * hp::MappingCollection).
```

Comment (luca-heltai) [inline include/deal.II/fe/fe_interface_values.h:1496]:
```suggestion
   * dominating finite element across the interface (only used if the
   * FEInterface object is initialised with an hp::FECollection, an
   * hp::QCollection, and possibly an hp::MappingCollection).
   * @param[in] mapping_index If left at its default, use the mapping within the
   * hp::MappingCollection passed to the constructor as given by the dominating
   * finite element across the interface (only used if the FEInterface object
   * is initialised with an hp::FECollection, an hp::QCollection, and possibly an
   * hp::MappingCollection).
   * @param[in] fe_index If left at its default, use the finite element within
   * the hp::FECollection passed to the constructor as given by the dominating
   * finite element across the interface (only used if the FEInterface object
   * is initialised with an hp::FECollection, an hp::QCollection, and possibly an
   * hp::MappingCollection).
```

Comment (luca-heltai) [reply include/deal.II/fe/fe_interface_values.h:1496]: same below.

**Review by luca-heltai — APPROVED** [review]
Except minor comments in the documentation, this looks good to me! Thanks @jppelteret !

**Comment by jppelteret** [comment]
Thanks for the review, @luca-heltai!

**Comment by luca-heltai** [comment]
Would you mind adding a ChangeLog?

**Comment by jppelteret** [comment]
> Would you mind adding a ChangeLog?

Of course. Perhaps you'd permit me to add it to the PR for #14594, since the testers are already running on this branch. I'll make a TODO there.

**Comment by luca-heltai** [comment]
We are getting 

```
Stacktrace
make[3]: *** [parameter_file_2.debug/output] Error 1
make[2]: *** [CMakeFiles/a-framework.parameter_file_2.debug.test.dir/all] Error 2
make[1]: *** [CMakeFiles/a-framework.parameter_file_2.debug.test.dir/rule] Error 2
make: *** [a-framework.parameter_file_2.debug.test] Error 2
Test a-framework/parameter_file_2.debug: RUN
===============================   OUTPUT BEGIN  ===============================
Scanning dependencies of target dummy.debug
[ 66%] Building CXX object CMakeFiles/dummy.debug.dir/dummy.cc.o
[ 83%] Linking CXX executable dummy.debug
[ 83%] Built target dummy.debug
Scanning dependencies of target a-framework.parameter_file_2.debug.test
[100%] Generating parameter_file_2.debug/output
a-framework/parameter_file_2.debug: BUILD successful.
a-framework/parameter_file_2.debug: RUN failed. ------ Return code 126
a-framework/parameter_file_2.debug: RUN failed. ------ Result: /home/dealii/build/tests/a-framework/parameter_file_2.debug/failing_output
a-framework/parameter_file_2.debug: RUN failed. ------ Partial output:
/home/dealii/build/share/deal.II/scripts/run_test.sh: line 88: /home/dealii/build/tests/a-framework/dummy.debug: Permission denied
CMakeFiles/a-framework.parameter_file_2.debug.test.dir/build.make:69: recipe for target 'parameter_file_2.debug/output' failed
CMakeFiles/Makefile2:461: recipe for target 'CMakeFiles/a-framework.parameter_file_2.debug.test.dir/all' failed
CMakeFiles/Makefile2:468: recipe for target 'CMakeFiles/a-framework.parameter_file_2.debug.test.dir/rule' failed
Makefile:285: recipe for target 'a-framework.parameter_file_2.debug.test' failed
a-framework/parameter_file_2.debug: ******    RUN failed    *******
===============================    OUTPUT END   ===============================
```
But I think this is unrelated with this PR. @tjhei ?

**Comment by tjhei** [comment]
Can you try with a clean build directory? This is related to what @tamiko did in #14615.

**Comment by jppelteret** [comment]
I went ahead and rebased the branch, and added a changelog while I was at it.

**Pull request merged**